File: src/single_camera/src/camera_subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include <sensor_msgs/msg/image.hpp>
#include <cv_bridge/cv_bridge.h>
#include <opencv2/highgui/highgui.hpp>


using std::placeholders::_1;

class CameraSubscriberNode : public rclcpp::Node{
public:
    CameraSubscriberNode() : Node("SingleCameraSubscriberNode"){
         image_sub = this->create_subscription<sensor_msgs::msg::Image>(
      "image", 10, std::bind(&CameraSubscriberNode::camera_callback, this, _1));
    }

    void camera_callback(const sensor_msgs::msg::Image::SharedPtr msg){
        cv_bridge::CvImagePtr cv_ptr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8);
        image = cv_ptr->image;
        cv::imshow("Image", image);
        cv::waitKey(1);       
  }
  
private:
    rclcpp::Subscription<sensor_msgs::msg::Image>::SharedPtr image_sub;
    cv::Mat image;
};

int main(int argc, char ** argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<CameraSubscriberNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
}

File: src/single_camera_python_subscriber/single_camera_python_subscriber/camera_subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import Int8
from sensor_msgs.msg import Image
from cv_bridge import CvBridge
import cv2


class MyNode(Node):       
    def __init__(self):
        super().__init__('SingleCameraSubscriber')
       
        self.image_sub = self.create_subscription(Image, 'image', self.listener_callback, 10)

    def listener_callback(self, msg):
        """ Callback for image subscriber """
        bridge = CvBridge()

        image = bridge.imgmsg_to_cv2(msg, 'bgr8')
        
        cv2.imshow("Output image", image)
        cv2.waitKey(1)
            
def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

